### PR TITLE
Features/body classification lightning module + training pipeline

### PR DESCRIPTION
**Pull request by jasonadriann**
# Description

Adding lightning module for body part classification using ResNet-18 + the training pipeline

Source:
* [Lightning Module](https://lightning.ai/docs/pytorch/stable/common/lightning_module.html)
* [VinDr-BodyPartXR](https://vindr.ai/datasets/bodypartxr) dataset
* [ResNet-18](https://arxiv.org/abs/1512.03385v1) architecture

New Features:

* Adding ResNet-18 architecture using torch module
* Create lightning module for bodypart classification using ResNet-18
* Create Hydra config for datamodule + model
* Create testing functions for training pipeline

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

The pr has been tested using [pytest](https://docs.pytest.org/en/7.4.x/) and dummy dataset that are being built with [tmp_path_factory](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html). Training is done using Hydra and Lightning.

**Test Configuration**:

* Firmware version:
* Hardware: IdeaPad Gaming 3 + RTX3050

# Checklist:

- [x] Did you make sure title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/models/components/resnet18.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                in_channels, out_channels)
+
+        return nn.Sequential(
+            BasicBlock(in_channels, out_channels,
+                       identity_downsample=identity_downsample, stride=stride),
+            BasicBlock(out_channels, out_channels)
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.conv1(x)
```
Comment (haritsahm) [inline src/models/components/resnet18.py:65]: missing docstring

File: tests/conftest.py [hunk 2]
```diff
@@ -212,3 +212,40 @@ def cfg_physenet(cfg_physenet_global, tmp_path) -> DictConfig:
     yield cfg
 
     GlobalHydra.instance().clear()
+
+
+@pytest.fixture(scope='package')
+def cfg_bodyresnet_global() -> DictConfig:
+    with initialize(version_base='1.3', config_path='../configs'):
```
Comment (haritsahm) [inline tests/conftest.py:219]: missing docstring

File: tests/conftest.py [hunk 3]
```diff
@@ -212,3 +212,40 @@ def cfg_physenet(cfg_physenet_global, tmp_path) -> DictConfig:
...
+            cfg.data.pin_memory = False
+            cfg.extras.print_config = False
+            cfg.extras.enforce_tags = False
+            cfg.logger = None
+
+    return cfg
+
+
+@pytest.fixture(scope='function')
+def cfg_bodyresnet18(cfg_bodyresnet_global, tmp_path) -> DictConfig:
```
Comment (haritsahm) [inline tests/conftest.py:242]: missing docstring

File: tests/test_resnet18_models.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import pytest
+import torch
+
+from src.models.components.resnet18 import ResNet18
+
+
+def test_resnet18():
+    model = ResNet18(1, 5)
```
Comment (haritsahm) [inline tests/test_resnet18_models.py:8]: missing docstring

File: tests/test_train_bodypartxr.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    bodypartxr_dummy_val_dataset_dir,
+)
+from tests.helpers.run_if import RunIf
+from tests.helpers.run_sh_command import run_sh_command
+
+startfile = 'src/train.py'
+overrides = ['logger=[]']
+
+
+def test_train_bodypartxr_fast_dev_run(cfg_bodyresnet18, bodypartxr_dummy_test_dataset_dir, bodypartxr_dummy_train_dataset_dir, bodypartxr_dummy_val_dataset_dir):
```
Comment (haritsahm) [inline tests/test_train_bodypartxr.py:20]: missing docstrings

**Review by haritsahm — CHANGES_REQUESTED** [review]
Please add necessary docstrings

**Comment by jasonadriann** [comment]
Adding several necessary docstrings. Please kindly review the changes

**Review by haritsahm — COMMENTED** [review]
Will add more features for testing

**Review by haritsahm — APPROVED** [review]
LGTM

**Pull request merged**